framework/auth/views.py

```diff
@@ -185,6 +188,27 @@ def auth_logout(redirect_url=None):
     return resp
 
 
+def auth_email_logout(token, user):
+    """When a user is adding an email or merging an account, add the email to the user and log them out.
+    """
+    redirect_url = web_url_for('auth_login') + '?existing_user={}'.format(escape_html(user.email))
```

Comment:
As suggested in a previous comment, you should URI-encode the email (not escape it). You can use `urllib.quote_plus` for this.